File: src/main/java/com/ukm/programkerja/service/UkmService.java
package com.ukm.programkerja.service;

import com.ukm.programkerja.dto.request.UkmRequest;
import com.ukm.programkerja.dto.response.UkmResponse;

import java.util.List;

public interface UkmService {
    UkmResponse create(UkmRequest request);
    UkmResponse update(Long id, UkmRequest request);
    void delete(Long id);
    UkmResponse getById(Long id);
    List<UkmResponse> getAll();
}

File: frontend/src/services/ukmService.js
import api from './api'

export default {
  getAll() {
    return api.get('/ukm')
  },
  getById(id) {
    return api.get(`/ukm/${id}`)
  },
  create(data) {
    return api.post('/ukm', data)
  },
  update(id, data) {
    return api.put(`/ukm/${id}`, data)
  },
  delete(id) {
    return api.delete(`/ukm/${id}`)
  },
}
